### PR TITLE
Require Node.js 12 or later & switch CI to GitHub Actions

### PR DESCRIPTION
**Pull request by eemeli**
Really we could update the minimum dependency to Node.js 14 as everything older than that is out of support.

The CI tests are here set up to run on a 4x3 matrix of systems, covering Node.js 12, 14, 16, and 18 on Linux, Mac & Windows.

As the action doesn't get run for the PR actually adding it (that's controlled by the config on the default branch), a test run is available here: https://github.com/eemeli/source-map/actions/runs/3196070788

### CODE REVIEW

**Comment by hildjj** [comment]
+1 for testing & supporting node [14,16,18] only.  That's a semver_major change, in my opinion.

**Review by hildjj — APPROVED** [review]

**Comment by hildjj** [comment]
Version numbers are cheap. :)
My opinion is start on 1.0 now, then use 2.0 (or whatever is next at that point) for the refactor.

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        node-version: [12, 14, 16, 18]
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm install --legacy-peer-deps
```
Comment (jkrems) [inline .github/workflows/nodejs.yml:24]: Is it worth it to create a `package-lock.json` and use `npm ci` here to ensure reproducible builds?

Comment (eemeli) [reply .github/workflows/nodejs.yml:24]: Eventually, yes. I'm hoping to add a separate PR adding a `package-lock.json` file, once at least this and #463 have been merged.

**Review by jkrems — APPROVED** [review]

**Review by ochameau — APPROVED** [review]
Thanks for jumping on this right away, this looks good.
See my inline comments.

File: .github/workflows/nodejs.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      - run: npm test
+
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+      - run: npm install --legacy-peer-deps
```
[the remark this answers is not in the document]
Comment (eemeli) [reply .github/workflows/nodejs.yml:33]: Yes, and yes.

**Comment by eemeli** [comment]
Tested the merge on my fork, looks pretty green: https://github.com/eemeli/source-map/actions/runs/3295739934

**Pull request merged**